SpatialGDK/Documentation/content/porting-unreal-project-to-gdk.md

```diff
@@ -113,63 +128,68 @@ etc...
     >
-    > In the future, we expect to achieve parity with Unreal networking in terms of iteration speed and stability, thus removing the need for a networking mode switch.
-1. Open the game's uproject file in the Editor and navigate to *Edit > Project Settings... > SpatialOS Unreal GDK > Toolbar*.
+    > You can easily switch between networking modes using this checkbox. As the SpatialOS Unreal GDK is in alpha, switching back to Unreal's networking may be a viable way for debugging any issues you may be having and may speed up your iteration times. However, in Unreal default networking mode you lose access to the multi-server features of the GDK, which may lead to erratic behavior.
 
-1. Tag all replicated classes with the UCLASS tag `SpatialType`:
-    ``` 
-    UCLASS(SpatialType) 
-    ```
-1. Locate your game's `GameState`, and make it a public singleton:
+
+1. To provide context to SpatialOS on how to replicate [singleton classes]({{urlRoot}}/content/singleton-actors) (such as GameState and GameMode) we have added a `UCLASS` [specifier](https://docs.unrealengine.com/en-US/Programming/UnrealArchitecture/Reference/Classes/Specifiers) which provides said context. The next steps detail how to do this.
+   
+1. If your game's '`GameState`' is a cpp file, locate it and mark it as a public singleton by modifying the UCLASS specifier as shown:
     ```
     UCLASS(SpatialType=Singleton)
     ```
-1. Locate your game's `GameMode`, and make it a private singleton:
+    If your game's '`GameState`' is a blueprint then you will need to open the full blueprint editor for that GameState and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton` to the `Spatial Description` textbox.
+
+1. If your game's '`GameMode`' is a cpp file, locate it and mark it as a private singleton by modifying the UCLASS specifier as shown:
     ```
     UCLASS(SpatialType=(Singleton,ServerOnly))
     ```
+    If your game's '`GameMode`' is a blueprint then you will need to open the full blueprint editor for that GameMode and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton,ServerOnly` to the `Spatial Description` textbox.
+
+1. Marking these singleton classes as `Spatial Type` will cause spatial schema to be generated for said classes when pressing the `Schema` button in the `SpatialOSGDKToolbar`. We now need to mark the generated schema components as [streaming queries](https://docs.improbable.io/reference/13.3/shared/worker-configuration/bridge-config#streaming-queries) for your UnrealWorker worker configuration, this ensures your UnrealWorkers always have these singletons checked out. The generated schema is located at `<ProjectRoot>\spatial\schema\improbable\unreal\generated`.
```

Comment:
Indent me!

---
* Move into active voice
* Avoid "said"
* Instruction rather than invitation ("you do" rather than "we do"
* Use "server-worker" and "client-worker" because that's what the SpatialOS docs call it.
* This goes into "check out" 
* This goes into "workers" 
* This goes into "schema"
* This goes into "streaming queries"
4 huge new concepts in one sentence - this is a mind blow situation for someone porting a game to SpatialOS for the first time. I think we need to add info about these concepts at the top of this section.
->

"1. Marking these classes as `Spatial Type` enables them to work with SpatialOS the GDK generates  SpatialOS schema for them when you select the **Schema** button in the SpatialOS toolbar.
The resulting generated schema components need to be marked as streaming queries for your server-worker and client-worker (that is `UnrealWorker`) configuration. This ensures your server-workers and client-workers always have these components checked out. The generated schema is located at `<ProjectRoot>\spatial\schema\improbable\unreal\generated`."